### PR TITLE
add support for struct field init shorthand

### LINKED ISSUE #65: Field Init Shorthand (like rust)
Hi @aep,

I hope you are doing well. I am curious for your thoughts about supporting _field init shorthand parameters_, [much like Rust](https://doc.rust-lang.org/book/ch05-01-defining-structs.html#using-the-field-init-shorthand-when-variables-and-fields-have-the-same-name) [1] for ZZ.

\[1]: https://doc.rust-lang.org/book/ch05-01-defining-structs.html#using-the-field-init-shorthand-when-variables-and-fields-have-the-same-name

### PR DESCRIPTION
**Pull request by jwerle**
Fixes #65

### CODE REVIEW

**Comment by aep** [comment]
lgtm, sorry for the late response.

CI failure is unrelated.

could you add a test?

File: src/zz.pest [hunk 1]
```diff
@@ -234,7 +234,7 @@ takeref         = { "&"  ~ expr_to_precedence_2}
 arrayaccess     = _{ expr ~ "]" }
 cast            = { "as" ~ "<" ~ anon_type ~ ">" ~ expr_to_precedence_2 | "(" ~ anon_type ~ ")" ~ expr_to_precedence_2}
 
-struct_init_field = { ident ~ ":" ~ expr }
+struct_init_field = { (expr | ident) ~ ":"? ~ expr? }
```
Comment (aep) [inline src/zz.pest:237]: actually this needs to be ( ":" ~ expr )    ?  otherwise it would be valid to add a colon but no value

Comment (jwerle) [reply src/zz.pest:237]: I tried that, but the parser now throws an error for regular `key: value` field definitions

Comment (aep) [reply src/zz.pest:237]: Probably because it's circular. 

Thinking of it, expr is actually wrong, since you can actually only use expression::name.

**Comment by jwerle** [comment]
I added a test

File: src/zz.pest (hunk 1 above)
Comment (jwerle) [reply src/zz.pest:237]: hmm so what do you suggest?

Comment (aep) [reply src/zz.pest:237]: sorry wasnt near a computer yesterday.  it should probably be 

    { (ident ~ ( ":" ~ expr) ? }

which means the first ident is either the key or a local name to be looked up by abs
for the second case (when expr is missing)  the parser needs to wrap the ident in an ast::Expression::Name so abs correctly looks it up in context

Comment (aep) [reply src/zz.pest:237]: pseudo code:

name = part.next().unwrap();
expr = if let Some(e) = part.next() {
    parse_expr(e)
} else {
   Expression::Name(name)
}

Comment (jwerle) [reply src/zz.pest:237]: thanks for the tip! still green to Pest and emerging on Rust. Diving into this now

**Comment by aep** [comment]
lgtm, thanks!

**Comment by jwerle** [comment]
:raised_hands:

**Pull request merged**